prediction_market_agent/agents/microchain_agent/functions.py

```diff
@@ -51,13 +54,7 @@ def example_args(self) -> list[str]:
         return []
 
     def __call__(self) -> list[str]:
-        # Get the 5 markets that are closing soonest
-        markets: list[AgentMarket] = OmenAgentMarket.get_binary_markets(
-            filter_by=FilterBy.OPEN,
-            sort_by=SortBy.CLOSING_SOONEST,
-            limit=5,
-        )
-
+        markets: list[OmenAgentMarket] = get_omen_binary_markets()
         market_questions_and_prices = []
         for market in markets:
             market_questions_and_prices.append(market.question)
```

Comment:
Maybe just a style preference, but I would prefer having some kind of tuple structure, like [(market_price, market_question)] instead of a list containing different item types. What do you think?

Reply:
Agreed, is there a reason why we can't use some Pydantic model here?

This seems like it could work:

```python
from pydantic import BaseModel


class MicroMarket(BaseModel):
    question: str
    p_yes: float

    def __str__(self) -> str:
        return f"'{self.question}' with probability of yes: {self.p_yes:.2%}"


mm = MicroMarket(question="Will it rain tomorrow?", p_yes=0.7)

prompt = f"""
You have following markets:

- {mm}
"""

print(prompt)
```

prints:

```bash
> python test.py                                                                                                                             [9:44:12]

You have following markets:

- 'Will it rain tomorrow?' with probability of yes: 70.00%
```

Reply:
Nice, will do this!

I think this is still a starting point though, and we might want to do some thinking/experimenting how we want the api to look like for the agent. We could expose a bunch more things (opening time, closing time, liquidity, ...), either as extra fields of this pydantic class, or as separate `Functions` the agent can call.

My hunch is that separate functions makes more sense for token efficiency, but need to check.

Also, we could expose just the market Id here, and have the agent pass that into the other getter functions, so that we don't need `get_omen_market_by_title` here https://github.com/gnosis/prediction-market-agent-tooling/pull/172/files.